### PR TITLE
feat: parse golang str radix

### PR DESCRIPTION
**Pull request by yuanbohan**

### CODE REVIEW

File: src/util/number.rs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    s.parse::<f64>()
+        .map_err(|_| format!("ParseFloatError. {s} can't be parsed into f64"))
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_parse_golang_str_radix() {
```
Comment (waynexia) [inline src/util/number.rs:62]: Well, this case is duplicated to the above doc test. We can choose one to keep.

File: src/util/number.rs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        assert_eq!(15_f64, parse_golang_str_radix("0xf").unwrap());
+        assert_eq!(-15_f64, parse_golang_str_radix("-0xf").unwrap());
+        assert_eq!(7_f64, parse_golang_str_radix("07").unwrap());
+        assert_eq!(-7_f64, parse_golang_str_radix("-07").unwrap());
+        assert_eq!(7_f64, parse_golang_str_radix("7").unwrap());
+        assert_eq!(-7_f64, parse_golang_str_radix("-7").unwrap());
```
Comment (waynexia) [inline src/util/number.rs:68]: what about `+07`/`+7`

Comment (waynexia) [reply src/util/number.rs:68]: and `abcdefgh`/`0xabcdefgh`

File: src/util/number.rs (hunk 1 above)
Comment (yuanbohan) [reply src/util/number.rs:62]: :ok_hand:

File: src/util/number.rs (hunk 2 above)
Comment (yuanbohan) [reply src/util/number.rs:68]: :ok_hand:

**Review by waynexia — APPROVED** [review]
LGTM

**Pull request merged**